components/sidebar/selection.ts

```diff
@@ -1,13 +1,48 @@
+import { EntryType, Pattern, TenureType } from "@/lib/types"
 import { proxy, useSnapshot } from "valtio"
 
 type SelectionStore = {
   patternNames: string[]
+  entryType: string | undefined
+  tenureTypes: TenureType[]
```

Comment:
I actually spent a bit of time trying to change these arrays to be `Set`s which is essentially what they are and how we're treating them. This would simplify the logic of adding / removing.

When I did this, I was getting issues with React updating and re-rendering changes. I could click accordion items, log out the correct state, but would need to open and close the sidebar to see a change.

Parked for now and I'll take another look later 👍